tools/kompos/src/view.ts

```diff
@@ -388,13 +388,72 @@ function constructChannelBpMenu(fileNode, tag: string, tpl: string) {
   });
 }
 
+function enableCreatePromisePairClicks(fileNode) {
```

Comment:
a lot of code duplication introduce here, could that please be avoided, also for existing code?